frontend/apps/reader/modules/readerbookmark.lua

```diff
@@ -59,6 +75,58 @@ function ReaderBookmark:addToMainMenu(menu_items)
             end,
         }
     end
+    menu_items.bookmarks_items_per_page = {
+        text = _("Bookmarks per page"),
+        keep_menu_open = true,
+        callback = function()
+            local SpinWidget = require("ui/widget/spinwidget")
+            local curr_perpage = G_reader_settings:readSetting("bookmarks_items_per_page") or self.bookmarks_items_per_page_default
+            local items = SpinWidget:new{
+                width = math.floor(Screen:getWidth() * 0.6),
+                value = curr_perpage,
+                value_min = 6,
+                value_max = 24,
+                default_value = self.bookmarks_items_per_page_default,
+                title_text =  _("Bookmarks per page"),
+                callback = function(spin)
+                    G_reader_settings:saveSetting("bookmarks_items_per_page", spin.value)
+                end
+            }
+            UIManager:show(items)
+        end
+    }
+    menu_items.bookmarks_items_font_size = {
+        text = _("Bookmark font size"),
+        keep_menu_open = true,
+        callback = function()
+            local SpinWidget = require("ui/widget/spinwidget")
+            local curr_perpage = G_reader_settings:readSetting("bookmarks_items_per_page") or self.bookmarks_items_per_page_default
+            local default_font_size = Menu.getItemFontSize(curr_perpage)
+            local curr_font_size = G_reader_settings:readSetting("bookmarks_items_font_size") or default_font_size
+            local items_font = SpinWidget:new{
+                width = math.floor(Screen:getWidth() * 0.6),
+                value = curr_font_size,
+                value_min = 10,
+                value_max = 72,
+                default_value = default_font_size,
+                title_text =  _("Bookmark font size"),
+                callback = function(spin)
+                    G_reader_settings:saveSetting("bookmarks_items_font_size", spin.value)
+                end
+            }
+            UIManager:show(items_font)
+        end,
+    }
+    menu_items.bookmarks_items_show_more_text = {
+        text = _("Reduce bookmark font size to show more text"),
```

Comment:
Is the "show more text" part necessary?
```suggestion
        text = _("Smaller bookmark font size"),
```

Reply:
I think so, it's supposed to only go smaller when the text is long enough and it helps showing more/all of it.

Reply:
I misunderstood then.

Something like `Auto(matically) shrink bookmark font size to fit more text` perhaps?

Reply:
![image](https://user-images.githubusercontent.com/24273478/106927438-72524a80-6712-11eb-9f5c-3a5062e12152.png)
vs:
![image](https://user-images.githubusercontent.com/24273478/106927488-81d19380-6712-11eb-8d48-72c4196dbfdf.png)